src/Core/Helpers.cs

```diff
@@ -60,21 +60,37 @@ private static HttpRequestMessage CreateProxyHttpRequest(this HttpContext contex
 
             var requestMessage = new HttpRequestMessage();
             var requestMethod = request.Method;
+
+            // Copy the request headers.
+            if (requestMessage.Content != null)
+            {
+                foreach (var header in request.Headers)
+                {
+                    if (!requestMessage.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray()))
+                    {
+                        requestMessage.Content?.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray());
+                    }
+                }
+            }
+
             if (!HttpMethods.IsGet(requestMethod) &&
                 !HttpMethods.IsHead(requestMethod) &&
                 !HttpMethods.IsDelete(requestMethod) &&
                 !HttpMethods.IsTrace(requestMethod))
             {
-                var streamContent = new StreamContent(request.Body);
-                requestMessage.Content = streamContent;
+                if (request.HasFormContentType)
+                {
+                    requestMessage.Content = new FormUrlEncodedContent(
+                        request.Form
+                            .Select(x => new KeyValuePair<string, StringValues>(x.Key, x.Value)).ToDictionary(x => x.Key, x => x.Value.ToString()));

```

Comment:
Can you make this a little more readable?  Maybe...

```csharp
requestMessage.Content = new FormUrlEncodedContent(
    request.Form
        .Select(x => new KeyValuePair<string, StringValues>(x.Key, x.Value))
        .ToDictionary(x => x.Key, x => x.Value.ToString())
);
```

Reply:
Also, can't you skip the `Select` clause?  I think you could just call `ToDictionary`, though I'd have to look at my intellisense probably.

Reply:
Yep, you have a right